### DISCUSSION TITLE
Linkable Results

### DISCUSSION THREAD

**Discussion by edthamm** [grafana/grafana#37621] [General]
| As | Engineer|
|---|---|
|I want to | send a link to a single log line to other engineers|
| so that | I can have a focused discussion about an issue|

---
**Acceptance Criteria**

1. I can generate a link that will show a single log line
2. This link will open up a page showing the single selected line
3. That page contains a link that will lead me to the query that generated the line (basically 'back to query')

---
**Background**

We are transitioning from Kibana+ES to Grafana+Loki. In this transition we found that it is common within our Org that engineers send each other links to specific log lines to focus a discussion around an issue.

We found out that this is not possible in Grafana. For us however this would be very handy to have.
Any idea if and how that could be made a reality?

**Comment by saada** [comment]
We've found a workaround that isn't perfect.

1. Set a date range by clicking on the pagination link on the right of the log output
2. Add a filter using a unique value in your log like a timestamp `{namespace="myapp"} |= "1637094889.7608848"`
3. Click the Share button on top of the Explore page

Ideally it would be a share button that shows up when a log line is expanded. Grafana already guesses the `timeNs` value at runtime. We could leverage that same value to filter out the time range as well as the filter for the log?

**Reply by edthamm** [reply]
Sorry for the late feedback.

Thanks a bunch for the workaround. I communicated that to our users and they feel that it works, but only kind of. Having that button on expand would be a really sweet thing.
Would that be `tsNs` in the detected fields?
I have 0 expertise in FE work so I have no clue how much work that would be. All I can say it that it would help a lot.

**Comment by gelicia** [comment]
This is available https://grafana.com/docs/grafana/latest/explore/logs-integration/#copy-link-to-log-line, closing discussion
